topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -316,107 +350,68 @@ Dynamic destinations allow you to write custom python code to dispatch jobs base
 >    {% endraw %}
 >    ```
->    {: data-commit="Deploy tool destinations config file"}
+>    {: data-commit="Importing TPV shared database via job conf"}
+>
+> Note how TPV allows the file to be imported directly via its http url. As many local and remote rule files as necessary can be combined, with rule files specified later overriding
+> any previously specified rule files. The TPV shared database does not define destinations, only cores and mem settings, as well as any required environment vars.
+> Take a look at the shared database of rules and note that some tools have very large recommended memory settings, which may or may not be available within your local cluster.
+> Nevertheless, you may still wish to execute these tools with memory adjusted to suit your cluster's capabilities. 
 >
-> 3. We need to update Galaxy's job configuration to use this rule. Open `templates/galaxy/config/job_conf.yml.j2` and add a DTD destination.
->    Also, comment out or remove the previous `<tool>` definition for the `testing` tool, and replace it with a mapping to the dtd destination like so:
+> 2. Edit your `files/galaxy/config/tpv_rules_local.yml` and make the following changes.
 >
 >    {% raw %}
 >    ```diff
->    --- a/group_vars/galaxyservers.yml
->    +++ b/group_vars/galaxyservers.yml
->    @@ -66,11 +66,14 @@ galaxy_job_config:
->           dynamic_admin_only:
->             runner: dynamic
->             function: admin_only
->    +      dtd:
->    +        runner: dynamic
->    +        type: dtd
->       tools:
->         - class: local # these special tools that aren't parameterized for remote execution - expression tools, upload, etc
->           environment: local_env
->         - id: testing
->    -      environment: dynamic_admin_only
->    +      environment: dtd
->     
->     galaxy_config:
->       galaxy:
+>    --- a/files/galaxy/config/tpv_rules_local.yml
+>    +++ b/files/galaxy/config/tpv_rules_local.yml
+>    @@ -30,7 +30,10 @@ destinations:
+>       slurm:
+>         inherits: singularity
+>         runner: slurm
+>    -    max_accepted_cores: 16
+>    +    max_accepted_cores: 24
+>    +    max_accepted_mem: 256
+>    +    max_cores: 16
+>    +    max_mem: 128
+>         params:
+>           native_specification: --nodes=1 --ntasks=1 --cpus-per-task={cores}
+>
 >    {% endraw %}
 >    ```
->    {: data-commit="Configure dtd in job conf"}
+>    {: data-commit="TPV clamp max cores and mem"}
+>
+> These changes indicate that the destination will accept jobs that are up to `max_accepted_cores: 24` and `max_accepted_mem: 256`. However, once accepted, it will forcibly clamp it down to 16 and 128 at most using the `max_cores` and `max_mem` clauses.
+> If the tool does not exceed the specified limit, it will run happily. If not, it will be clamped down to the desired range. This allows even the largest resource requirement in the shared database to be accomodated.
```

Comment:
Probably a good idea to explicitly state what happens to any tool that defines more than `max_accepted_mem/cores`. Will it just be rejected?

Reply:
Yes. I'll add that clarification.